src/elements/content-sidebar/SidebarNav.js

```diff
@@ -87,87 +107,122 @@ const SidebarNav = ({
             focusPrompt();
         }
     };
+    const { icon: CustomTabIcon, index: customTabIndex = 0, title: customTabTitle, navButtonProps } = customTab || {};
+    const customTabId = customTab && typeof customTab.id === 'string' ? customTab.id.trim() : '';
+    const hasCustomTab = !!customTabId;
+    const hasBoxAICustomTab = customTabId === 'boxai';
+
+    // Configuration-driven button definitions
+    const getButtonConfigs = () => {
+        const configs = [];
+
+        // BoxAI button (always first)
+        if (hasBoxAI && !hasBoxAICustomTab) {
+            configs.push({
+                key: 'boxai',
+                id: 'boxAI',
+                view: SIDEBAR_VIEW_BOXAI,
+                analyticsTarget: SIDEBAR_NAV_TARGETS.BOXAI,
+                testId: 'sidebarboxai',
+                tooltip: showOnlyBoxAINavButton ? boxAIDisabledTooltip : intl.formatMessage(messages.sidebarBoxAITitle),
```

Comment:
@coderabbitai In functional components, I'm used to seeing lines like...

```
const { formatMessage } = useIntl();

...

formatMessage(messages.sidebarBoxAITitle)
```

What's the equivalent in class components, where intl is passed as a prop?